### DISCUSSION TITLE
过期：懒人包 2023V2

### DISCUSSION THREAD

**Discussion by hooke007** [hooke007/mpv_PlayKit#260] [回收]
# 版本专帖

- 02/12 文档站已同步至mpv主线 

## 懒人包

正如上期的宣告，v1版很短命。<s>上游不积极的话预计会存在一到两个季度</s>

- 01/28: portable_config/profiles.conf 里写错了一行。重传以修复，已下的用户也不用重新下，将此行的 `[ontop_playback]`
https://github.com/hooke007/MPV_lazy/blob/f0294a5f589593ca7ce3fafb0dece5d32426cbd4/portable_config/profiles.conf#L14-L15
修改为 `[vsync_auto]` 即可
https://github.com/hooke007/MPV_lazy/blob/c51b9aa1971a3fbfd1940f1beaa3508b5522aac1/portable_config/profiles.conf#L14-L15
- 03/11: portable_config/vs/waifu2x_cuda.vpy 中存在错误，使得1080p的动态超分无法进行，如有需求参考此提交 https://github.com/hooke007/MPV_lazy/commit/7b3edee2b6998c02e35891f380596f9c17c420dc 进行手动修复。

## 简易包

终版已发布，终止维护。

<hr />
链接 <a href='https://github.com/hooke007/MPV_lazy/releases/tag/20230127'>mpv-lazy-20230127</a>

链接 <a href='https://github.com/hooke007/mpv.net_CM/releases/tag/6032re'>mpvnet-CM-obs</a>

**Comment by zvvv007** [comment]
https://s1.ax1x.com/2023/01/29/pSaT16s.png 出现爆内存的情况，可复现，具体是全屏观看一定时间后，返回窗口模式，内存就会一直增长，直到mpv卡死

**Reply by hooke007** [reply]
我无法复现这个问题

**Reply by zvvv007** [reply]
> 我无法复现这个问题

多次全屏，退出全屏后最小化，再全屏看一段时间可以复现

**Comment by arasakaxx** [comment]
想請問爲什麽我開了rife補幀都沒有效果 mstool可以正常使用 是我有什麽前置作業沒做嗎 https://imgur.com/a/eRcQj1a

**Reply by hooke007** [reply]
看控制台消息

**Reply by arasakaxx** [reply]
看完了他寫找不到 可是我確認過了有在那個路徑啊 https://imgur.com/a/g90TtVd

**Reply by hooke007** [reply]
这个报错就是找不到目标文件，另外路径不要有中文

**Reply by hooke007** [reply]
额外的，不要放在od里，这不是一个正常的路径

**Reply by arasakaxx** [reply]
有了 把它移到其他地方后 感謝你的幫忙

**Comment by chenyk98** [comment]
请问有没有办法修改 usoc 中 `open-file` 的 path 路径 ？
现在定位是在 %User% 下 ，比如能不能定位到驱动器列表？

**Reply by hooke007** [reply]
https://github.com/hooke007/MPV_lazy/blob/35b6041c21ada257af5799ff4e9562f84f3c753e/portable_config/script-opts/uosc.conf#L132

**Comment by lbcmiao** [comment]
现有的右键菜单不跟随鼠标，而且需要点击才能进入二级菜单，操作手感和普通右键菜单差异太大了。感觉更像是给电视机遥控使用的。

**Reply by hooke007** [reply]
osd只能渲染在界面内，固定位置渲染是最佳解决方案。

**Comment by HovardHe** [comment]
播完暂停，无法自动连播；列表循环会单集循环，随机播放会播完暂停

**Reply by hooke007** [reply]
建议完整看完说明。分清楚播放列表和文件列表

**Comment by Rayrisss** [comment]
你好，请问如图设置下播放高分辨率视频时冻屏，拖动进度条才能播放怎么解决呢？部分高分辨率视频甚至无法使spv启动
<img src="https://user-images.githubusercontent.com/124227504/216226036-92a9be29-2ccc-44f9-a3de-91f790f9eaa1.jpg" width=400> <img src="https://user-images.githubusercontent.com/124227504/216226031-645651ad-569a-4072-a26d-768ed1a626b5.jpg" width=400>
<img src="https://user-images.githubusercontent.com/124227504/216226026-d24e21d5-f440-47df-b83e-362054b784ce.jpg" width=400>

**Reply by hooke007** [reply]
建议不要东拼西凑抄配置，里面充斥了大量意义不明的操作。

**Comment by zorgzola** [comment]
三代笔记本mpv.exe崩溃，找不到原因，其他程序运行正常。系统很干净，核显HD3000、独显Geforce 610M。pot+madvr运行正常。
windows日志提示
由于下列某个原因，Windows 无法访问文件 : 网络连接、存储文件的磁盘或安装在此计算机上的存储 驱动程序有问题；或者磁盘丢失。 由于上述错误，Windows 关闭了程序 mpv
另一个错误日志
错误应用程序名称: mpv.exe，版本: 2.0.0.0，时间戳: 0x63d427c0
错误模块名称: mpv.exe，版本: 2.0.0.0，时间戳: 0x63d427c0
异常代码: 0xc000001d
错误偏移量: 0x00000000026e20b9
错误进程 ID: 0x0x998
错误应用程序启动时间: 0x0x1D93708F22EF5CE
错误应用程序路径: D:\Program Files\mpv-lazy\mpv.exe
错误模块路径: D:\Program Files\mpv-lazy\mpv.exe
报告 ID: 0c3ac866-e225-4ab1-8d1b-6a985bc3bbad
错误程序包全名: 
错误程序包相对应用程序 ID:

**Reply by hooke007** [reply]
我没理解错三代是指3700那一代？建议不要忽略说明文档。你需要手动替换内核

**Reply by zorgzola** [reply]
抱歉，说错了，是intel 3代cpu Ivy Bridge，i5-3230M，核显HD4000。
你需要手动替换内核没找到方法。
多谢大佬回答

**Reply by hooke007** [reply]
下载非v3版的mpv.exe
https://github.com/shinchiro/mpv-winbuild-cmake

**Reply by zorgzola** [reply]
感谢，可以运行了。

**Comment by SuperBlackWednesday** [comment]
```
vf-append = vapoursynth="~~/vs/mvtools-2x.vpy"
```
上个版本 这么设置能打开就2倍帧率播放 这么版本这么设置就不行  这么要怎么破？

**Reply by SuperBlackWednesday** [reply]
因为是默认播放器   上个版本  我双击视频图标  直接就开始播放了  而且进去就是2倍帧率   这么版本需要右击屏幕选择2倍帧率  多个步骤啊

**Reply by hooke007** [reply]
你先把文件名打对了。
这个改变至少2022的终版已经是这么处理了 https://github.com/hooke007/MPV_lazy/releases/tag/20221112#:~:text=标准化命名格式
你的上个版本必然不是v1，如果是那必然是你在用旧版本的脚本。

**Comment by Wirtz-7** [comment]
Anime4K_Upscale_GAN_x2_M和NNEDI3无效果，统计信息界面可以看到着色器工作但画面无变化https://imgur.com/a/dOcXbPs

**Reply by hooke007** [reply]
效果是有的，自行截图ab

**Reply by Wirtz-7** [reply]
截图测试了一下，开关着色器画面应该是没有变化的，我个人测试默认的几个着色器，Anime4K_Restore_CNN_L和Adaptive Sharpen都能看出明显改善画面，其他的几个好像都不能生效

**Reply by hooke007** [reply]
> 应该是没有变化的

如何论证是否完全相同可以算hash，算ssim。相同才能证明没有变化。

> Anime4K_Restore_CNN_L和Adaptive Sharpen都能看出明显改善画面

Restore 和 Adaptive Sharpen 本身就是强观感。另外Adaptive Sharpen那也不叫“改善画质”。锐化的本质是失真——牺牲画质改变观感。

> 其他的几个好像都不能生效

在我眼里无疑他们都生效，可是你看不出来。
类似于音质，不是所有人有能力精确区分无损和320kbps，也不是所有人能感受这些放大算法差异。这也是事实，可以通过大量后天训练和劣化源品质（越垃圾的源越能体现出差异）去提高能力或者提高辨识度。

**Reply by Wirtz-7** [reply]
谢谢，麻烦大佬了，元宵节快乐

**Comment by LYX-Rain** [comment]
请问如何快速将一个文件夹内所有视频加到播放列表内？

**Reply by hooke007** [reply]
https://github.com/hooke007/MPV_lazy/wiki/2_%E8%AE%BE%E7%BD%AE%E6%96%87%E4%BB%B6%E5%A4%B9%EF%BC%88%E4%BE%BF%E6%90%BA%E5%BC%8F%EF%BC%89#%E5%A4%96%E7%BD%AE%E8%84%9A%E6%9C%AC%E8%B7%AF%E5%BE%84-

**Comment by MrPinku** [comment]
我想问一下字幕设置，这里默认是sub-ass-force-margins = yes
我看仓库的在线版系列手册这里是说“如果字幕是ASS格式，在有黑边的情况下，启用将顶部字幕和字幕放进黑边。默认： no”
但是我现在遇到一个新状况，我看4:3的视频，全屏的时候特效字幕会跑到左边的黑边去：
<img src="https://user-images.githubusercontent.com/79267206/217725467-00be3dbc-f8e0-45b7-9da1-dd803be7217e.png" width=400> <img src="https://user-images.githubusercontent.com/79267206/217725496-862b1691-29e3-436e-9065-da728023e3ad.png" width=400>
我也喜欢把上下的字幕放到上下黑边的功能，有设置可以保留这个功能的同时字幕不会跑到左右的黑边去的办法吗？

**Reply by hooke007** [reply]
基于快捷键的方案；
此项可以在运行时更改；
或者pad滤镜

**Reply by MrPinku** [reply]
我用上了快捷键 cycle sub-ass-force-margins，效果很好。
另外我试了一下做了一个预设，默认打开，4：3视频关闭。
```
[ass_margins]
 profile-desc = 非4:3视频ASS字幕自动放进黑边
 profile-cond = p["video-params/aspect"]<1.5
 profile-restore = copy
 sub-ass-force-margins = no
```

**Comment by bheith** [comment]
默认设置直接用全屏后会疯狂闪烁，同时estimated数值异常，窗口化后就恢复正常，如何解决？
![QQ图片20230219140207](https://user-images.githubusercontent.com/99644063/219931938-44d1bfbf-e6fd-49b4-ab80-756b749c25a9.png)
![QQ图片20230219140215](https://user-images.githubusercontent.com/99644063/219931944-5fc09973-3b82-46fc-a8d9-e88e4d860546.png)

**Reply by hooke007** [reply]
faq 垂直同步

**Reply by bheith** [reply]
> faq 垂直同步

已解决 感谢

**Comment by tomriddle1234** [comment]
右键菜单文字极小，字体也模糊，在42寸和27寸的4K显示器上都看不清楚，请问是怎么回事，支不支持High DPI？调整OSD部分字体大小设置，无效，OSD部分设置只影响文件名，对播放时间线按钮和右键菜单无效。

**Reply by hooke007** [reply]
先看完说明

**Reply by tomriddle1234** [reply]
> 先看完说明

实在没找到哪里关于这个问题的说明，mpv.conf里面设置吗？刚发现放大全屏以后，右键菜单字体变大，但是还是看不清。

**Reply by tomriddle1234** [reply]
感谢找到了，mpv.conf里面一行， hidpi-window-scale = yes

**Reply by hooke007** [reply]
<img width="969" alt="截屏2023-02-20 09 45 46" src="https://user-images.githubusercontent.com/41094733/220070726-c0eaa3f5-d0e6-46eb-ad31-02971e7fda88.png">

https://github.com/hooke007/MPV_lazy/wiki/0_FAQ#高分屏的界面元素过小

**Comment by wli189** [comment]


**Reply by hooke007** [reply]
你觉得我有能力在一无所知的情况下回答原因吗。另外用no-config测试最新版mpv，如果依然存在，向mpv报告它的问题。

**Reply by wli189** [reply]
话说有log记录吗，好像找不到

**Reply by hooke007** [reply]
https://github.com/hooke007/MPV_lazy/blob/fa132abc61325713953aefbe07e0ffa85bdfd166/portable_config/mpv.conf#L8

**Reply by wli189** [reply]
好像播本地就不会崩溃

**Reply by wli189** [reply]
然后emby就必崩
[mpv-lazy.log](https://github.com/hooke007/MPV_lazy/files/10800111/mpv-lazy.log)

**Reply by hooke007** [reply]
是emby崩还是mpv崩，我不使用emby

**Reply by wli189** [reply]
是当emby外部播放器时候，mpv崩，好像和osd有关，不知道osd是啥玩意。

**Reply by hooke007** [reply]
如果原版一样崩，那你应该问emby，和我的修改无关

**Comment by wathper** [comment]
请问三月份有计划发布新的发行版吗？

**Comment by wli189** [comment]
开启gpu next播放hdr会紫屏
[mpv-lazy.log](https://github.com/hooke007/MPV_lazy/files/10917169/mpv-lazy.log)
![image](https://user-images.githubusercontent.com/32810655/223618441-45e8bd45-4e3c-453f-9368-1ef260ee95df.png)

[   8.528][d][vo/gpu-next/libplacebo] LUT cache invalidated, regenerating..
[   8.532][d][vo/gpu-next/libplacebo] LUT cache invalidated, regenerating..
[   8.532][e][vo/gpu-next/libplacebo] Failed rendering image!
[   8.532][e][vo/gpu-next] Failed rendering frame!

**Reply by hooke007** [reply]
gpu-next用最新版测试

**Reply by wli189** [reply]
mpv 版本 mpv-x86_64-v3-20230305-git-291e05e

**Reply by wli189** [reply]
回退0226就没事了

**Reply by wli189** [reply]
![image](https://user-images.githubusercontent.com/32810655/223619564-1b5ca19e-5be0-4158-b9d4-518a91a33401.png)
还有这个，是为什么，只有hdr报错

**Reply by hooke007** [reply]
这不是我的问题，向mpv回报它的问题

**Reply by wli189** [reply]
好

**Comment by MichaelChou98** [comment]
第一次接触这个软件，请问是直接下载标准版使用就行了吗，还是需要下其他版本体验更好？我的是N卡

**Reply by hooke007** [reply]
包是按需扩展的，没有什么体验更好的说法

**Comment by superciv** [comment]
懒人包mpv.conf中 target-prim=display-p3，但查阅手册并没有这个选项，是否应改写为dci-p3?
附手册原文：
--target-prim=<value>
Specifies the primaries of the display. Video colors will be adapted to this colorspace when ICC color management is not being used. Valid values are:
auto
Disable any adaptation, except for atypical color spaces. Specifically, wide/unusual gamuts get automatically adapted to BT.709, while standard gamut (i.e. BT.601 and BT.709) content is not touched. (default)
bt.470m
ITU-R BT.470 M
bt.601-525
ITU-R BT.601 (525-line SD systems, eg. NTSC), SMPTE 170M/240M
bt.601-625
ITU-R BT.601 (625-line SD systems, eg. PAL/SECAM), ITU-R BT.470 B/G
bt.709
ITU-R BT.709 (HD), IEC 61966-2-4 (sRGB), SMPTE RP177 Annex B
bt.2020
ITU-R BT.2020 (UHD)
apple
Apple RGB
adobe
Adobe RGB (1998)
prophoto
ProPhoto RGB (ROMM)
cie1931
CIE 1931 RGB (not to be confused with CIE XYZ)
dci-p3
DCI-P3 (Digital Cinema Colorspace), SMPTE RP431-2
v-gamut
Panasonic V-Gamut (VARICAM) primaries
s-gamut
Sony S-Gamut (S-Log) primaries

**Reply by hooke007** [reply]
手册不全

**Comment by abearinalaska** [comment]
使用mpv播放视频时，不论是立体声还是5.1、7.1声道，感觉音量都偏小，尤其是人声和背景声，调大音量后，音乐声、伴奏声或爆炸音效等又非常突兀。请问有没有方案可以将mpv的声音调整到potplayer的效果？

**Reply by hooke007** [reply]
最简化的解决方案就是
`af-append=dynaudnorm` 或 `af-append=loudnorm`
https://github.com/hooke007/MPV_lazy/discussions/120#discussioncomment-2743978

**Reply by abearinalaska** [reply]
有效，感谢。个人使用loudnorm效果更好

**Comment by hooke007** [comment]
近期补充的超重ai上采样的 720px2 效果预览 (前一组图做了一点bilinear回缩)
性能要求 cugan > esrgan > waifu2x (预估实时1080p x2需要rtx3090) >>>>> anime4k > bilinear

bilinear (mpv默认)
<img width="400" alt="miku_bilinear" src="https://user-images.githubusercontent.com/41094733/227585512-852e4720-ab30-4972-9e7e-de58bd46a10e.png"> <img width="400" alt="miku2_bilinear" src="https://user-images.githubusercontent.com/41094733/227675672-161b5fa6-70d3-4e5f-b9e5-b286575fabf4.png">

a4k restore_cnn_L + upscale_gan_M
<img width="400" alt="miku_a4k" src="https://user-images.githubusercontent.com/41094733/227585516-80327656-84e8-464e-91a1-fdf98db36f3f.png"> <img width="400" alt="miku2_a4k" src="https://user-images.githubusercontent.com/41094733/227675654-0e16f845-7fb2-43d5-8db9-13ca501316da.png">

waifu2x upconv_7_anime_style_art_rgb 一级降噪
<img width="400" alt="miku_waifu2x" src="https://user-images.githubusercontent.com/41094733/227585525-73c1f74e-33d2-4747-b600-494a6592edab.png"> <img width="400" alt="miku2_waifu2x" src="https://user-images.githubusercontent.com/41094733/227675648-ee917b68-c145-418b-a731-8f4979deb900.png">

esrgan animevideov3
<img width="400" alt="miku_esrgan" src="https://user-images.githubusercontent.com/41094733/227585535-a0fdc337-467d-46de-a67d-6d21b8a794d2.png"> <img width="400" alt="miku2_esrgan" src="https://user-images.githubusercontent.com/41094733/227675659-ad02a383-3db0-4a9c-9a90-81a34e83f7c0.png">

cugan pro 不降噪
<img width="400" alt="miku_cugan" src="https://user-images.githubusercontent.com/41094733/227585545-ae6e2de7-2965-4887-aa95-32e5d6cfa249.png"> <img width="400" alt="miku2_cugan" src="https://user-images.githubusercontent.com/41094733/227675664-5653f4a3-9906-4567-8135-365db6acb11c.png">

**Comment by s3327445** [comment]
請問大家用過animejanai嗎？ 目前沒有中文說明 我一直裝失敗 engine無法生成

**Reply by s3327445** [reply]
再請問h大 esrgan及cugan怎麼使用的 我只找到waifu2x的vpy檔 以及設定的方法 是內部測試嗎？

**Reply by hooke007** [reply]
这些东西的示例，在上游和我的wiki/通用文档里都有写。我不可能每个部件每个脚本都写的很详细怎么一步步用（不然我得累死，举一反三只能你们自己用了）
另外最近的更新都是在上游分支上做的，不在主分支里

**Reply by s3327445** [reply]
感謝 我在找找看 好有趣

**Comment by TToSTTT** [comment]
想请问缓存和解复用器缓存的区别是什么？（看了文档没看懂）
我想实现播放流媒体视频（比如阿里网盘的资源）时，向前预先读取256M的内容，向后读取128M的内容，以便向前后跳转都不卡顿，这样写的配置正确吗？
![image](https://user-images.githubusercontent.com/125366009/227875731-9c6653ee-b96f-4e3a-81ef-ca127c90882f.png)

**Reply by hooke007** [reply]
你要的功能就是demuxer-*
--cache的默认值auto就够了，默认不对本地文件启用缓存，除非你有特殊用途（比如硬盘快挂了读取跟不上）

**Reply by TToSTTT** [reply]
好的，谢谢。

**Reply by ghost** [reply]
移动设备开本地文件内存缓存可以起到节能作用

**Comment by SuperBlackWednesday** [comment]
你好大神，请问右键菜单怎么变成英文？

**Reply by hooke007** [reply]
很怪异的需求。
改input conf里的对应文本，菜单内容怎么生成的在脚本说明文档里已经写了。
https://github.com/hooke007/MPV_lazy/discussions/186

**Comment by hooke007** [comment]
rife的时域TTA开了可能更好（动态画面的文字更少的扭曲）也可能更差（移动场景的密集线条更频繁的变形）

T_TTA关   vs.  T_TTA开（静止帧不明显，画面动起来会有闪烁感）
<img width=400 src="https://user-images.githubusercontent.com/41094733/228295177-d4da4d7a-3dbe-4d9c-83fe-41ed7c165010.png"> <img width=400 src="https://user-images.githubusercontent.com/41094733/228295184-678ed028-b6f0-4999-8fdc-b6bd65516816.png">

**Comment by hooke007** [comment]
https://github.com/hooke007/MPV_lazy/blob/main/portable_config/scripts/stats_plus.lua

想加功能但是整不明白，于是决定摆烂，但是改都改了好几天了删了可惜，<s>于是就传了这个除了界面语言有区别外没有任何区别的脚本。</s> 做了微量的增补调节（右图）
实在看stats上的词看不懂的可以用（除此以外我也想不到啥用处了）

<img width=400 src="https://user-images.githubusercontent.com/41094733/228971471-84793116-f140-41b5-a810-766142459778.jpg"> <img width=400 src="https://user-images.githubusercontent.com/41094733/229391787-9ea2dd4a-94d8-44ec-bfdd-ef8fe0255b55.jpg">

**Comment by hooke007** [comment]
着色器系列 nlmeans 当前被标记为与 nvidia 不兼容，不要使用。 guided 过于原始，不要使用。
